sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/KeyVaultTestBase.cs

```diff
@@ -28,15 +36,66 @@ internal SecretClient GetClient(TestRecording recording = null)
                     recording.InstrumentClientOptions(new SecretClientOptions())));
         }
 
+        public override void StartTestRecording()
+        {
+            base.StartTestRecording();
+
+            Client = GetClient();
+            VaultUri = new Uri(Recording.GetVariableFromEnvironment(AzureKeyVaultUrlEnvironmentVariable));
+        }
+
+        [TearDown]
+        public async Task Cleanup()
+        {
+            try
+            {
+                foreach (var cleanupItem in _secretsToCleanup)
+                {
+                    if (cleanupItem.Delete)
+                    {
+                        await Client.DeleteAsync(cleanupItem.Secret.Name);
```

Comment:
I'm a little lost here. If we are awaiting for Delete here, why do we check again that the secret goy marked as deleted with `WaitForDeletedSecret`?
Same question for purge

Reply:
KeyVault doesn't actually delete the secret immediately, it's state changes to "being deleted" but GetDeletedSecret would still return it and purge would fail during this time. Same for purge.

Reply:
I see ... like wait until we change the state.... got it.